File: Top 100 Codes/04-Important Codes related to Arrays/27-Finding equilibrium index of an array/Findingequilibriumindexofanarray.py
def find_equilibrium_indices(arr):
    total_sum = sum(arr)  # Compute total sum of the array
    left_sum = 0
    equilibrium_indices = []

    for i in range(len(arr)):
        # Right sum = total_sum - left_sum - arr[i]
        if left_sum == (total_sum - left_sum - arr[i]):
            equilibrium_indices.append(i)  # Store equilibrium index
        
        left_sum += arr[i]  # Update left sum for next iteration

    return equilibrium_indices

# Example usage
arr = [-7, 1, 5, 2, -4, 3, 0]
result = find_equilibrium_indices(arr)

# Output result
if result:
    print("Equilibrium indices:", result)
else:
    print("No equilibrium index found")

File: Top 100 Codes/04-Important Codes related to Arrays/27-Finding equilibrium index of an array/Findingequilibriumindexofanarray.cpp
#include <iostream>
using namespace std;

void findEquilibriumIndices(int arr[], int n) {
    int totalSum = 0, leftSum = 0;

    // Calculate total sum of the array
    for (int i = 0; i < n; i++) {
        totalSum += arr[i];
    }

    cout << "Equilibrium indices: ";
    bool found = false;

    // Traverse the array and check for equilibrium condition
    for (int i = 0; i < n; i++) {
        // Right sum = totalSum - leftSum - arr[i]
        if (leftSum == totalSum - leftSum - arr[i]) {
            cout << i << " ";
            found = true;
        }
        leftSum += arr[i]; // Update left sum
    }

    if (!found) {
        cout << "None";
    }
    cout << endl;
}

int main() {
    int arr[] = { -7, 1, 5, 2, -4, 3, 0 };
    int n = sizeof(arr) / sizeof(arr[0]);

    findEquilibriumIndices(arr, n);
    return 0;
}
